### ISSUE TITLE
Validation to terminate an option is not always correct

### ISSUE THREAD

**Issue by nomaed** [bitwisecook/vscode-iRule#6]
White it is true in various commands (like `switch`, `regsub`, `regexp`, so on) that if the first argument is a variable or a command (also a sequence of initial arguments), it might be evaluated as an option.

However in the case of an explicit value that is not an option, this warning is incorrect. There is no need to add `--` in this example:
![Screen Shot 2019-09-26 at 10 59 04](https://user-images.githubusercontent.com/9551921/65669716-bfb5a680-e04c-11e9-9f7a-70272cd2a07b.png)

Perhaps the warning should only be displayed if the argument can potentially be interpreted as an option?

Warning should be here:
```tcl
switch $val { ...
```
or here:
```tcl
switch [get_some_value] { ...
```

But not here:
```tcl
regsub {[^a-z]+} $val ...
```

**Comment by bitwisecook** [comment]
I've pushed a bunch of improvements to the diagnostics, but there's still work to do around this ticket.

**Comment by bitwisecook** [comment]
This should be improved some more but the real work is still off in the semantic highlighting/diagnostics branch

**Comment by bitwisecook** [comment]
I'm going to close this out, the irule style guide I wrote will be going up on devcentral this week and it addresses this.
